File: data/solver/test/test_cafe.py
import pytest
from cafe import Cafe

def test_cafe_init():
    cafe = Cafe("Starbucks", "New York", "123-456-7890")
    assert cafe.name == "Starbucks"
    assert cafe.location == "New York"
    assert cafe.contact == "123-456-7890"
    assert cafe.coffee_demand == {}

def test_set_coffee_demand():
    cafe = Cafe("Starbucks", "New York", "123-456-7890")
    cafe.set_coffee_demand("Cold Brew", 10)
    assert cafe.coffee_demand == {"Cold Brew": 10}

def test_get_coffee_demand():
    cafe = Cafe("Starbucks", "New York", "123-456-7890")
    cafe.set_coffee_demand("Cold Brew", 10)
    assert cafe.get_coffee_demand("Cold Brew") == 10
    assert cafe.get_coffee_demand("Espresso") == 0

def test_display_info(capsys):
    cafe = Cafe("Starbucks", "New York", "123-456-7890")
    cafe.set_coffee_demand("Cold Brew", 10)
    cafe.display_info()
    captured = capsys.readouterr()
    assert captured.out == "Name: Starbucks\nLocation: New York\nContact: 123-456-7890\nCoffee Demand:\n  - Cold Brew: 10\n"

def test_str():
    cafe = Cafe("Starbucks", "New York", "123-456-7890")
    assert str(cafe) == "Cafe(name=Starbucks, location=New York, contact=123-456-7890)"

File: data/solver/test/cpp/test_cafe.cpp
#include <iostream>
#include <sstream>
#include "Cafe.h"
#include "catch.hpp"

TEST_CASE("Test Cafe initialization") {
    Cafe cafe("Starbucks", "New York", "123-456-7890");
    REQUIRE(cafe.get_name() == "Starbucks");
    REQUIRE(cafe.get_location() == "New York");
    REQUIRE(cafe.get_contact() == "123-456-7890");
    REQUIRE(cafe.get_coffee_demand().empty());
}

TEST_CASE("Test set_coffee_demand") {
    Cafe cafe("Starbucks", "New York", "123-456-7890");
    cafe.set_coffee_demand("Cold Brew", 10);
    REQUIRE(cafe.get_coffee_demand()["Cold Brew"] == 10);
}

TEST_CASE("Test get_coffee_demand") {
    Cafe cafe("Starbucks", "New York", "123-456-7890");
    cafe.set_coffee_demand("Cold Brew", 10);
    REQUIRE(cafe.get_coffee_demand("Cold Brew") == 10);
    REQUIRE(cafe.get_coffee_demand("Espresso") == 0);
}

TEST_CASE("Test display_info") {
    Cafe cafe("Starbucks", "New York", "123-456-7890");
    cafe.set_coffee_demand("Cold Brew", 10);
    std::stringstream buffer;
    std::streambuf* old = std::cout.rdbuf(buffer.rdbuf());
    cafe.display_info();
    std::string output = buffer.str();
    std::cout.rdbuf(old);
    REQUIRE(output == "Name: Starbucks\nLocation: New York\nContact: 123-456-7890\nCoffee Demand:\n  - Cold Brew: 10\n");
}

TEST_CASE("Test Cafe to string") {
    Cafe cafe("Starbucks", "New York", "123-456-7890");
    REQUIRE(cafe.to_string() == "Cafe(name=Starbucks, location=New York, contact=123-456-7890)");
}